onnx/common/path.cc

```diff
@@ -9,99 +9,85 @@
 
 namespace ONNX_NAMESPACE {
 
-bool is_path_separator(char c) {
-  // Windows accept / as path separator.
-  if (k_preferred_path_separator == "\\") {
-    return c == '\\' || c == '/';
+#define PATH_JOIN(string_type, separator)                                         \
+  template <>                                                                     \
+  string_type path_join(const string_type& origin, const string_type& append) {   \
+    if (origin.find_last_of(separator) != origin.length() - separator.length()) { \
+      return origin + separator + append;                                         \
+    }                                                                             \
+    return origin + append;                                                       \
   }
 
-  return c == k_preferred_path_separator[0];
-}
-
-void normalize_separator(std::string& path) {
-  char preferred_sep = k_preferred_path_separator[0];
-  if (preferred_sep == '/') {
-    // Do nothing on linux.
-    return;
-  }
-
-  for (size_t i = 0; i < path.size(); i++) {
-    if (is_path_separator(path[i]) && path[i] != preferred_sep) {
-      path[i] = preferred_sep;
-    }
+#define CLEAN_RELATIVE_PATH(string_type, char_type, separator, dot)                                           \
+  template <>                                                                                                 \
```

Comment:
Would you please add a comment of what does this function do?

Reply:
Sounds good. Just added in path.h.